### PR TITLE
Add world loading

### PR DESCRIPTION
**Pull request by zp3x**
- [x] Level.dat: Load general world information (spawn point, time, dimension, etc.)
Chunk Management:
    - [x] Chunk Loading: Efficiently load and unload chunks based on player proximity or other criteria.
    Chunk Data:
        - [x] Block Data: Load block types, metadata, and tile entities.
        - [ ] Light Levels: Load skylight and block light data.
        - [ ] Entity Data: Load entities within the chunk, including their data.
        - [x] Biome Data: Load biome information for the chunk.

- [x] Spawn Chunks: Load initial chunks around the world spawn point.
- [ ] World Generation: Generate terrain for unloaded chunks (if necessary).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,7 @@
 /target
 configuration.toml
 features.toml
+*Cargo.lock
```
Comment (Snowiiii) [inline .gitignore:4]: Please leave Cargo.lock
https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html

File: pumpkin/src/protocol/client/config/mod.rs [hunk 2]
```diff
@@ -123,7 +123,8 @@ impl ClientPacket for CRegistryData {
         bytebuf.put_list::<Entry>(&self.entries, |p, v| {
             p.put_string(&v.entry_id);
             p.put_bool(v.has_data);
-            registry::write_single_dimension(p, -64, 320);
+
+            //registry::write_single_dimension(p, -64, 320);
```
Comment (Snowiiii) [inline pumpkin/src/protocol/client/config/mod.rs:127]: Why you comment that out?

File: pumpkin/src/world/block.rs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize)]
+pub struct Block {
+    pub x: i32,
+    pub y: i32,
+    pub z: i32,
+    pub id: i32,
```
Comment (Snowiiii) [inline pumpkin/src/world/block.rs:8]: For now an id is fine, I think in the future we should move to an enum

File: pumpkin/src/world/world.rs [hunk 4]
```diff
@@ -10,4 +10,19 @@ impl World {
             players: Vec::new(),
         }
     }
+
+    pub fn get_region_file(x: f32, z: f32) -> String {
+        let result: f32 = 7.0 / 32.0;
+        println!("{}", result.floor());
```
Comment (Snowiiii) [inline pumpkin/src/world/world.rs:16]: you can btw use `dbg!()` for debugging

File: pumpkin/src/world/block.rs (hunk 3 above)
Comment (zp3x) [reply pumpkin/src/world/block.rs:8]: Read the comment in chunk.rs!

File: pumpkin/src/world/chunk.rs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+impl WorldChunk {
+    //region is the folder with the region files in world it is: world/region, in world_nether it is: world_nether/DIM-1/region for example
+    pub fn load_chunk(region: &str, x: i32, z: i32) -> Self {
+        let mut blocks = Vec::new();
+
+        let region_file = format!("{}/{}", region, World::get_region_file(x as f32, z as f32));
+        let intern = World::get_intern_coords(x, z);
+        dbg!(intern);
+        dbg!(region_file.clone());
```
Comment (junderw) [inline pumpkin/src/world/chunk.rs:19]:
```suggestion
        let region_file = dbg!(format!("{}/{}", region, World::get_region_file(x as f32, z as f32)));
        let intern = dbg!(World::get_intern_coords(x, z));
```

**Pull request closed without merging**